pyiceberg/table/__init__.py

```diff
@@ -2931,14 +3161,52 @@ def _deleted_entries(self) -> List[ManifestEntry]:
         return []
 
 
-class OverwriteFiles(_MergingSnapshotProducer):
+class OverwriteFiles(_MergingSnapshotProducer["OverwriteFiles"]):
```

Comment:
Is there a reason why we are keeping this class over using DeleteFiles and FastAppend?

It looks like we have two different meaning / implementation of `overwrite` in Table/Transaction as opposed to `UpdateSnapshot`, where the former is producing a snapshot with the operation `delete` + `append` (and maybe another `append`) and the latter is producing a snapshot with the operation `overwrite`. I'm wondering if this would be confusing for the users

Reply:
Sounds good @Fokko - I've added a line item to discuss this at the next Monthly sync 👍

Reply:
Nevermind - I think I caught up with your latest decisions - so OVERWRITE snapshot will just be used to note that an existing snapshot was replaced/overwritten 👍